app/javascript/components/widgets/traffic/panel.jsx

```diff
@@ -93,6 +94,7 @@ const Traffic = () => (
 
       const columnSize = Math.floor(trafficCams.length / 2);
       const [firstColumn, secondColumn] = splitAt(columnSize, trafficCams);
+      const sourceUrls = uniq(trafficCams.map(cam => cam.sourceUrl));
```

Comment:
```js
const sourceUrls = uniq(trafficCams.map(cam => new URL(cam.url).origin));
```

Reply:
Bonus points for functional composition:
```
const getUniqueUrls = compose(uniq, map(cam => new URL(cam.url).origin);
...
          {getUniqUrls(trafficCams).map(url => {
            <Notice>{url}</Notice>
          ))}
```

Reply:
it just makes it easier for anyone to export the function later, if they need its output